src/saveload/vehicle_sl.cpp

```diff
@@ -812,66 +878,98 @@ SaveLoadTable GetVehicleDescription(VehicleType vt)
-
-		     SLE_VAR(Vehicle, sprite_cache.sprite_seq.seq[0].sprite, SLE_FILE_U16 | SLE_VAR_U32),
-		 SLE_CONDVAR(Vehicle, age,                   SLE_FILE_U16 | SLE_VAR_I32,   SL_MIN_VERSION,  SLV_31),
-		 SLE_CONDVAR(Vehicle, age,                   SLE_INT32,                   SLV_31, SL_MAX_VERSION),
-		     SLE_VAR(Vehicle, tick_counter,          SLE_UINT8),
-
-		 SLE_CONDVAR(DisasterVehicle, image_override,            SLE_FILE_U16 | SLE_VAR_U32,   SL_MIN_VERSION, SLV_191),
-		 SLE_CONDVAR(DisasterVehicle, image_override,            SLE_UINT32,                 SLV_191, SL_MAX_VERSION),
-		 SLE_CONDVAR(DisasterVehicle, big_ufo_destroyer_target,  SLE_FILE_U16 | SLE_VAR_U32,   SL_MIN_VERSION, SLV_191),
-		 SLE_CONDVAR(DisasterVehicle, big_ufo_destroyer_target,  SLE_UINT32,                 SLV_191, SL_MAX_VERSION),
-		 SLE_CONDVAR(DisasterVehicle, flags,                     SLE_UINT8,                  SLV_194, SL_MAX_VERSION),
+		    SLE_VAR(Vehicle, owner,                 SLE_UINT8),
+		    SLE_VAR(Vehicle, vehstatus,             SLE_UINT8),
+		SLE_CONDVAR(Vehicle, current_order.dest,    SLE_FILE_U8 | SLE_VAR_U16,    SL_MIN_VERSION,   SLV_5),
+		SLE_CONDVAR(Vehicle, current_order.dest,    SLE_UINT16,                   SLV_5, SL_MAX_VERSION),
+
+		    SLE_VAR(Vehicle, sprite_cache.sprite_seq.seq[0].sprite, SLE_FILE_U16 | SLE_VAR_U32),
+		SLE_CONDVAR(Vehicle, age,                   SLE_FILE_U16 | SLE_VAR_I32,   SL_MIN_VERSION,  SLV_31),
+		SLE_CONDVAR(Vehicle, age,                   SLE_INT32,                   SLV_31, SL_MAX_VERSION),
+		    SLE_VAR(Vehicle, tick_counter,          SLE_UINT8),
+
+		SLE_CONDVAR(DisasterVehicle, image_override,            SLE_FILE_U16 | SLE_VAR_U32,   SL_MIN_VERSION, SLV_191),
+		SLE_CONDVAR(DisasterVehicle, image_override,            SLE_UINT32,                 SLV_191, SL_MAX_VERSION),
+		SLE_CONDVAR(DisasterVehicle, big_ufo_destroyer_target,  SLE_FILE_U16 | SLE_VAR_U32,   SL_MIN_VERSION, SLV_191),
+		SLE_CONDVAR(DisasterVehicle, big_ufo_destroyer_target,  SLE_UINT32,                 SLV_191, SL_MAX_VERSION),
+		SLE_CONDVAR(DisasterVehicle, flags,                     SLE_UINT8,                  SLV_194, SL_MAX_VERSION),
 
 		SLE_CONDNULL(16,                                                           SLV_2, SLV_144), // old reserved space
 	};
+#if defined(_MSC_VER) && (_MSC_VER == 1915 || _MSC_VER == 1916)
+		return description;
+	}
+#endif
 
+	void GenericSaveLoad(Vehicle *v) const
+	{
+		if (v->type != VEH_DISASTER) return;
+		SlObject(v, this->GetDescription());
+	}
 
-	static const SaveLoadTable _veh_descs[] = {
-		_train_desc,
-		_roadveh_desc,
-		_ship_desc,
-		_aircraft_desc,
-		_special_desc,
-		_disaster_desc,
-		_common_veh_desc,
-	};
+	void Save(Vehicle *v) const override { this->GenericSaveLoad(v); }
+	void Load(Vehicle *v) const override { this->GenericSaveLoad(v); }
+	void FixPointers(Vehicle *v) const override { this->GenericSaveLoad(v); }
```

Comment:
I'm seeing these three return over and over again, might it be worth it to have an intermediate "super" class with virtual GenericSaveLoad, or maybe even a templated intermediate super class that requires the implementing sub class to provide a GenericSaveLoad? The same is already done with description, and it means less duplicating of these three lines all over the save load code.

Reply:
Later PRs mostly remove this; it is simply as a stop-gap. As I knew if I would to copy/paste the code and create identical functions I would get the complaints I have duplicated code :P This is the best I could do in the chain of PRs :)

But even after all this work there is plenty of room to refactor the code to have less code. But I left this out of this set of PRs, as reviewing this is already difficult enough :)